connectors/src/connectors/google_drive/lib.ts

```diff
@@ -58,14 +62,14 @@ export async function registerWebhooksForAllDrives({
 export async function ensureWebhookForDriveId(
   connector: ConnectorResource,
   driveId: string,
-  marginMs: number
+  renewWebhookStartingTime: number
 ): Promise<Result<string | undefined, Error>> {
   const webhook = await GoogleDriveWebhook.findOne({
     where: {
       connectorId: connector.id,
       driveId: driveId,
       expiresAt: {
-        [Op.gt]: new Date(new Date().getTime() + marginMs),
+        [Op.gt]: new Date(renewWebhookStartingTime),
```

Comment:
Are we sure of this `Op.gt` here? If expiresAt is behind us we do want to renew the webhook no?

Reply:
Damn yeah. I guess this + the race condition were the problem 💡

Reply:
Hmm IMHO this was an Op.gt here indeed. The code checks whether there is already a webhook that expires far enough in the future. If there isn't (normal case), it creates it. If there is (anormal case since we shouldn't have ran the webhook renewal) undefined is returned